stripe/src/main/java/com/stripe/android/cards/CardNumber.kt

```diff
@@ -1,60 +1,88 @@
 package com.stripe.android.cards
 
+import com.stripe.android.CardUtils
 import com.stripe.android.StripeTextUtils
 
-internal sealed class CardNumber(number: String) {
-    private val normalizedNumber = StripeTextUtils.removeSpacesAndHyphens(number).orEmpty()
-
+internal sealed class CardNumber {
     /**
      * A representation of a partial or full card number that hasn't been validated.
      */
-    class Unvalidated(number: String) : CardNumber(number)
+    internal data class Unvalidated internal constructor(
+        private val denormalizedNumber: String
+    ) : CardNumber() {
+        private val normalizedNumber = StripeTextUtils
+            .removeSpacesAndHyphens(denormalizedNumber)
+            .orEmpty()
 
-    /**
-     * Format a number based on its expected length
-     *
-     * e.g. `"4242424242424242"` with pan length `16` will return `"4242 4242 4242 4242"`;
-     * `"424242"` with pan length `16` will return `"4242 42"`;
-     * `"4242424242424242"` with pan length `14` will return `"4242 424242 4242"`
-     */
-    fun getFormatted(
-        panLength: Int = DEFAULT_PAN_LENGTH
-    ) = formatNumber(panLength)
+        fun validate(panLength: Int): Validated? {
```

Comment:
I'm curious, why do we take `panLength` as an argument rather than the `Unvalidated` instance's normalized length of `number`?

Reply:
We can't tell if a number is valid without knowing how long it's supposed to be (e.g. if a number's BIN is for a 19 digit number, we wouldn't want to count a 16 digit number as valid)